src/lib/utils/transformFilesToTreeData.tsx

```diff
@@ -0,0 +1,196 @@
+import type { TreeDataItem } from "@/components/ui/tree-view"
+import type {
+  IRenameFileProps,
+  IDeleteFileProps,
+} from "@/hooks/useFileManagement"
+import { isHiddenFile } from "@/components/ViewPackagePage/utils/is-hidden-file"
+import { File, Folder, MoreVertical, Pencil, Trash2 } from "lucide-react"
+import {
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuGroup,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from "@/components/ui/dropdown-menu"
+import { useToast } from "@/hooks/use-toast"
+
+type FileName = string
+
+interface TransformFilesToTreeDataProps {
+  files: Record<FileName, string>
+  currentFile: FileName | null
+  renamingFile: string | null
+  handleRenameFile: (props: IRenameFileProps) => { fileRenamed: boolean }
+  handleDeleteFile: (props: IDeleteFileProps) => { fileDeleted: boolean }
+  setRenamingFile: (filename: string | null) => void
+  onFileSelect: (filename: FileName) => void
+  onFolderSelect: (folderPath: string) => void
+  canModifyFiles: boolean
+  setErrorMessage: (message: string) => void
+  setSelectedFolderForCreation: (folder: string | null) => void
+}
+
+export const transformFilesToTreeData = ({
+  files,
+  currentFile,
+  renamingFile,
+  handleRenameFile,
+  handleDeleteFile,
+  setRenamingFile,
+  onFileSelect,
+  onFolderSelect,
+  canModifyFiles,
+  setErrorMessage,
+  setSelectedFolderForCreation,
+}: TransformFilesToTreeDataProps): TreeDataItem[] => {
+  const { toast } = useToast()
+  type TreeNode = Omit<TreeDataItem, "children"> & {
+    children?: Record<string, TreeNode>
+  }
```

Comment:
Move all the types to the top of the file, if not moving to a different file for types